pam/pam-module-transaction-dummy.go

```diff
@@ -0,0 +1,295 @@
+	}
+	return m.Env, nil
+}
+
+// GetUser is similar to GetItem(User), but it would start a conversation if
+// no user is currently set in PAM.
+func (m *PamModuleTransactionDummy) GetUser(prompt string) (string, error) {
+	if user, err := m.GetItem(pam.User); err != nil {
+		return "", err
+	} else if user != "" {
+		return user, nil
+	}
+
+	resp, err := m.StartStringConv(pam.PromptEchoOn, prompt)
+	if err != nil {
+		return "", err
+	}
+
+	return resp.Response(), nil
+}
+
+// SetData allows to save any value in the module data that is preserved
+// during the whole time the module is loaded.
+func (m *PamModuleTransactionDummy) SetData(key string, data any) error {
+	if m.Data == nil {
+		m.Data = make(map[string]any)
+	}
+
+	m.Data[key] = data
+	return nil
+}
+
+// GetData allows to get any value from the module data saved using SetData
+// that is preserved across the whole time the module is loaded.
+func (m *PamModuleTransactionDummy) GetData(key string) (any, error) {
+	if m.Data == nil {
+		return nil, pam.NoModuleData
+	}
+
+	data, found := m.Data[key]
+	if found {
+		return data, nil
+	}
+	return nil, pam.NoModuleData
+}
+
+// DummyStringResponse is a simple implementation of pam.StringConvResponse.
+type DummyStringResponse struct {
+	style   pam.Style
+	content string
+}
+
+// Style returns the conversation style of the DummyStringResponse.
+func (s DummyStringResponse) Style() pam.Style {
+	return s.style
+}
+
+// Response returns the string response of the DummyStringResponse.
+func (s DummyStringResponse) Response() string {
+	return s.content
+}
+
+// DummyBinaryResponse is an implementation of pam.BinaryConvResponse.
+type DummyBinaryResponse struct {
+	ptr pam.BinaryPointer
+}
+
+// Style returns the response style for the response, so always BinaryPrompt.
+func (b DummyBinaryResponse) Style() pam.Style {
+	return pam.BinaryPrompt
+}
+
+// Data returns the response native pointer, it's up to the protocol to parse
+// it accordingly.
+func (b DummyBinaryResponse) Data() pam.BinaryPointer {
+	return b.ptr
+}
+
+// Decode decodes the binary data using the provided decoder function.
+func (b DummyBinaryResponse) Decode(decoder pam.BinaryDecoder) (
+	[]byte, error) {
+	if decoder == nil {
+		return nil, errors.New("nil decoder provided")
+	}
+	return decoder(b.ptr)
+}
+
+// StartStringConv starts a text-based conversation using the provided style
+// and prompt.
+func (m *PamModuleTransactionDummy) StartStringConv(style pam.Style, prompt string) (
+	pam.StringConvResponse, error) {
+	switch style {
+	case pam.BinaryPrompt:
+		return nil, pam.NewTransactionError(errors.New("Binary style is not supported"),
+			pam.ConvErr)
+	}
+
+	res, err := m.StartConv(pam.NewStringConvRequest(style, prompt))
+	if err != nil {
+		return nil, err
+	}
+
+	if stringRes, ok := res.(pam.StringConvResponse); ok {
+		return stringRes, nil
+	}
+
+	return nil, pam.NewTransactionError(
+		errors.New("Can't convert to pam.StringConvResponse"), pam.ConvErr)
```

Comment:
Same comment about the "happy path" here. The expected return is the response, so it's more idiomatic to do:
```Go
stringRes, ok := res.(pam.StringConvResponse)
if !ok {
    return nil, pam.NewTransactionError(errors.New("Can't convert to pam.StringConvResponse"), pam.ConvErr)
}
return stringRes, nil
```